s3_scrubber/src/lib.rs

```diff
@@ -77,9 +80,13 @@ impl Display for NodeKind {
 impl S3Target {
     pub fn with_sub_segment(&self, new_segment: &str) -> Self {
         let mut new_self = self.clone();
-        let _ = new_self.prefix_in_bucket.pop();
-        new_self.prefix_in_bucket =
-            [&new_self.prefix_in_bucket, new_segment, ""].join(&new_self.delimiter);
+        if new_self.prefix_in_bucket.is_empty() {
```

Comment:
```suggestion
        if new_self.prefix_in_bucket.pop().is_none() {
```

Reply:
Hmm, I think this is a bit easier to read the way it is, albeit more wordy.